backends/redis.go

```diff
@@ -0,0 +1,118 @@
+package backends
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/go-redis/redis"
+)
+
+type RedisBackend struct {
+	client *redis.Client
+}
+
+func NewRedisBackend(connString string) (*RedisBackend, error) {
+	u, _ := url.Parse(connString)
```

Comment:
Fun fact: I did something almost identical at first but got sick of writing regular expressions. I went full lazy bastard and leaned into `url.Parse` after that.